########################################
# Motion Planning (MoveGroupInterface) #
########################################

# Describes benchmark parameters (independant of profilers)
benchmark_config:
  parameters:
    runs: 5                         # Number of trials for each query
    name: "MotionPlanning MGI"      # Default ""    | Overridden by ROS Param /benchmark/name
    visualize: false                # Default false | Overridden by ROS Param /benchmark/visualize
    output_file: ""                 # Default ""    | Overridden by ROS Param /benchmark/output_file

# Describes config for building motion planning pair-wise queries
profiler_config:

  # OPTIONAL Override 'requests' with moveit_msgs/MotionPlanRequest respective field/s
  requests_override:
    allowed_planning_time: 5.0
    num_planning_attempts: 1

  # Resource moveit_msgs/MotionPlanRequest
  requests:
    - name: jc
      resource: package://moveit_benchmark_suite_resources/requests/bbt/panda/jc.yaml
    - name: jc_pcp
      resource: package://moveit_benchmark_suite_resources/requests/bbt/panda/jc_pcp.yaml

  # Resource moveit_config (robot)
  robot:
    name: panda
    resources:
      urdf: /robot_description
      srdf: /robot_description_semantic
      kinematics: /robot_description_kinematics
      joint_limits: /robot_description_planning/joint_limits

  # Resource moveit_msgs/PlanningScene
  scenes:
    - name: empty_scene
      resource: package://moveit_benchmark_suite_resources/scenes/empty/empty.urdf.xacro
    - name: primitive_scene
      resource: package://moveit_benchmark_suite_resources/scenes/bbt/panda/primitive.urdf.xacro
    # - name: mesh_scene
    #   resource: package://moveit_benchmark_suite_resources/scenes/bbt/panda/mesh_16k.urdf.xacro

  collision_detectors:
    - FCL

  # Resource moveit_config (planning pipeline)
  planning_pipelines:
    - name: ompl
      # Override/Extend moveit_msgs/MotionPlanRequest pipeline_id field
      resource: /move_group/planning_pipelines/ompl
      # OPTIONAL Override/Extend moveit_msgs/MotionPlanRequest planner_id field
      parameters:
        planners:
          - RRTConnect
